sdk/core/scheduler/main.cc

```diff
@@ -642,3 +661,20 @@ int multiwaiter_wait(Timeout           *timeout,
 		return 0;
 	});
 }
+
+#ifdef SCHEDULER_ACCOUNTING
+[[cheri::interrupt_state(disabled)]] uint64_t thread_elapsed_cycles_idle()
```

Comment:
They're disabled for paranoia.  idleThreadCycles shouldn't be updated in the middle of this call: if the thread is preempted then it is runnable and the idle thread is never scheduled if there's a runnable thread, so this can't be modified concurrently with a read.  But I don't want to have to think about that, so interrupts are disabled.